tools/autograd/derivatives.yaml

```diff
@@ -716,6 +716,9 @@
 - name: soft_margin_loss_forward(Tensor self, Tensor target, bool size_average, bool reduce)
   self: soft_margin_loss_backward(grad, self, target, size_average, reduce)
 
+- name: relu(Tensor self)
```

Comment:
nit: You don't need to specify a derivative for `relu` if you define `relu` with `threshold`:
```
Tensor relu(const Tensor & self) {
  return self.threshold(0, 0);
}
```
but then it'll look different from the definition of `relu_` (there isn't a `threshold_` function I think)

Reply:
It seems that `relu` with `clamp_min` is slightly faster than `threshold` too though?

Reply:
Good point. That's interesting:

![image](https://user-images.githubusercontent.com/5652049/37303162-fc977aec-2603-11e8-92c4-c1efd2194cba.png)
(threshold on left, clamp_min on right (as cmaxValue))

I don't think the conditional should slow down the code by as much as you're seeing but... that can be an investigation for another day